compiler/rustc_expand/src/expand.rs

```diff
@@ -2179,6 +2179,42 @@ impl<'a, 'b> MutVisitor for InvocationCollector<'a, 'b> {
         self.visit_node(node)
     }
 
+    fn filter_map_where_predicate(
+        &mut self,
+        mut pred: ast::WherePredicate,
+    ) -> Option<ast::WherePredicate> {
+        if self.cx.ecfg.features.cfg_attribute_in_where() {
+            while let Some((attr, pos, _derives)) = self.take_first_attr(&mut pred) {
+                match attr.name_or_empty() {
+                    sym::cfg => {
+                        if !self.expand_cfg_true(&mut pred, attr, pos).0 {
+                            return None;
+                        }
+                    }
+                    _ => continue,
+                }
+            }
+            let spans = pred
+                .attrs
+                .iter()
+                .filter(|attr| attr.name_or_empty() != sym::cfg)
+                .map(|attr| attr.span)
+                .collect::<Vec<_>>();
+            if !spans.is_empty() {
+                self.cx.dcx().emit_err(UnsupportedAttributesInWhere { span: spans.into() });
+            }
```

Comment:
This should go in `rustc_passes::check_attr`, which runs after every expansion has been done.